provider/resources.go

```diff
@@ -454,14 +455,72 @@ func preConfigureCallbackWithLogger(credentialsValidationRun *atomic.Bool, gcpCl
 //go:embed cmd/pulumi-resource-gcp/bridge-metadata.json
 var metadata []byte
 
+// A predicate function that always returns true.
+func always[T any](T) bool { return true }
+
+// fixEmptyLabels applies a state edit to fix
+// https://github.com/pulumi/pulumi-gcp/issues/2372.
+func fixEmptyLabels(_ context.Context, req shimv2.PlanStateEdit) (cty.Value, error) {
+	inputs, ok := resource.PropertyPath{"labels"}.Get(resource.NewProperty(req.NewInputs))
+	if !ok {
+		return req.PlanState, nil
+	}
+
+	labels := resource.FromResourcePropertyValue(inputs)
+	if !labels.IsMap() {
+		// No labels found, so we don't need to correct
+		return req.PlanState, nil
+	}
+
+	// fixOutputLabels fixes falsely computed values inherited from "labels".
+	fixOutputLabels := func(output cty.Value) cty.Value {
+		if !output.Type().IsMapType() {
+			return output
+		}
+		m := output.AsValueMap()
+		for k, v := range m {
+			if v.IsKnown() {
+				// If v is known, then it is not falsely computed. No
+				// action is needed.
+				continue
+			}
+			label, ok := labels.AsMap()[k] // labels is in the Pulumi namespace
+			if !ok || label.IsComputed() || !label.IsString() {
+				// If we didn't inherit label from "labels" or the label
```

Comment:
Does the issue not happen with empty labels in defaultLabels/effectiveLabels?

Reply:
It did, and that was addressed.